src/main/java/net/snowflake/ingest/streaming/internal/ChannelCache.java

```diff
@@ -53,6 +106,20 @@ void addChannel(SnowflakeStreamingIngestChannelInternal<T> channel) {
     return this.cache.entrySet().iterator();
   }
 
+  /**
+   * Returns an iterator over the (table, channels) in this map, filtered by the given table name
+   * set
+   *
+   * @param tableNames the set of table names to filter
+   * @return
+   */
+  Iterator<Map.Entry<String, ConcurrentHashMap<String, SnowflakeStreamingIngestChannelInternal<T>>>>
+      iterator(Set<String> tableNames) {
```

Comment:
Do you expect a lot of places to be doing this filtered iteration? If its just one callsite you could just do the if check at the callsite? Esp since the iterator is already exposed in the previous method, we don't need to add a method here for every kind of usage of the iterator.

Reply:
Use `entrySet()` and `keySet()`.